rotkehlchen/data_import/importers/bittrex.py

```diff
@@ -40,52 +52,160 @@ def _consume_trades(
+                    formatstr='%m/%d/%Y %I:%M:%S %p',
+                    location='Bittrex order history import',
+                ),
+                location=Location.BITTREX,
+                base_asset=asset_from_bittrex(csv_row['MARKET']),
+                quote_asset=quote_asset,
+                trade_type=TradeType.BUY if csv_row['ORDERTYPE'] == 'LIMIT_BUY' else TradeType.SELL,  # noqa: E501
+                amount=deserialize_asset_amount(csv_row['FILLED']),
+                rate=deserialize_price(csv_row['LIMIT']),
+                link=f'Imported bittrex trade {csv_row["UUID"]} from csv',
+            )
 
     @staticmethod
     def _consume_deposits_or_withdrawals(
-            csv_row: dict[str, Any], timestamp_format: str = '%Y-%m-%d %H:%M:%S.%f',
+            csv_row: dict[str, Any],
+            file_type: BittrexFileType,
+            timestamp_format: str = '%Y-%m-%d %H:%M:%S.%f',
     ) -> AssetMovement:
         """
         Use Deposit and Withdrawal entries to generate an AssetMovement object.
         May raise:
         - KeyError
         - DeserializationError
         """
-        asset = asset_from_bittrex(csv_row['Currency'])
-        return AssetMovement(
-            location=Location.BITTREX,
-            category=AssetMovementCategory.DEPOSIT if csv_row['Type'] == 'DEPOSIT' else AssetMovementCategory.WITHDRAWAL,  # noqa: E501
-            address=deserialize_asset_movement_address(csv_row, 'Address', asset),
-            transaction_id=get_key_if_has_val(csv_row, 'TxId'),
-            timestamp=deserialize_timestamp_from_date(
-                date=csv_row['Date'],
-                formatstr=timestamp_format,
-                location='Bittrex tx history import',
-            ),
-            asset=asset,
-            amount=deserialize_asset_amount_force_positive(csv_row['Amount']),
-            fee_asset=asset,
-            fee=Fee(ZERO),
-            link='Imported from bittrex CSV file',
-        )
+        if file_type == BittrexFileType.DEPOSITS_AND_WITHDRAWALS:
+            asset = asset_from_bittrex(csv_row['Currency'])
+            return AssetMovement(
+                location=Location.BITTREX,
+                category=AssetMovementCategory.DEPOSIT if csv_row['Type'] == 'DEPOSIT' else AssetMovementCategory.WITHDRAWAL,  # noqa: E501
+                address=deserialize_asset_movement_address(csv_row, 'Address', asset),
+                transaction_id=get_key_if_has_val(csv_row, 'TxId'),
+                timestamp=deserialize_timestamp_from_date(
+                    date=csv_row['Date'],
+                    formatstr=timestamp_format,
+                    location='Bittrex tx history import',
+                ),
+                asset=asset,
+                amount=deserialize_asset_amount_force_positive(csv_row['Amount']),
+                fee_asset=asset,
+                fee=Fee(ZERO),
+                link='Imported from bittrex CSV file',
+            )
```

Comment:
can't you abstract this logic a bit? seems that you can create a function that formats the movements and provide arguments like `transaction_id_key`, `address_key`, ...